### PR TITLE
feat: 208: added capability to reopen a past notification

### PR DESCRIPTION
**Pull request by lasmil**
- added ability to reopen a past notification
- closes: https://github.com/nearform/github-snooze-chrome-extension/issues/208

### CODE REVIEW

File: src/api/chrome.js [hunk 1]
```diff
@@ -137,6 +137,26 @@ export const removeSnooze = async (userId, snooze) => {
...
+ * @param {object} snooze object
+ * @returns the list of snoozes after the update
+ */
+ export const updateSnooze = async (userId, snooze) => {
+  const snoozeList = await getSnoozeList(userId)
+  const updatedList = snoozeList.map(s => {
+    if (s.id === snooze.id) {
+      return snooze
+    }
+    return s
```
Comment (ilteoood) [inline src/api/chrome.js:152]:
```
return currentSnooze.id === wantedSnooze.id ? wantedSnooze : currentSnooze
```

File: src/components/SnoozeItem.js [hunk 2]
```diff
@@ -74,18 +76,30 @@ function SnoozeItem({ user, snooze, onDelete }) {
...
+              <DialogFormButton
+                title="When do you want to be notified?"
+                description={url}
+                placeholder="Select or type your own value"
+                onConfirm={(notifyDate) => onUpdateSnooze({ notifyDate, snooze })}
+                size="small"
+                iconButtonComponent={<EditIcon />}
+              />
+            }
+            {status === SNOOZE_STATUS_DONE ? (
```
Comment (ilteoood) [inline src/components/SnoozeItem.js:90]: what if you make a dedicated component for this block?

File: src/api/chrome.js (hunk 1 above)
Comment (lasmil) [reply src/api/chrome.js:152]: On it

File: src/components/SnoozeItem.js (hunk 2 above)
Comment (lasmil) [reply src/components/SnoozeItem.js:90]: On it

File: src/api/chrome.js (hunk 1 above)
Comment (lasmil) [reply src/api/chrome.js]: Done

File: src/App.js [hunk 3]
```diff
@@ -58,6 +58,35 @@ const App = () => {
     setSnoozeList(updatedSnoozeList)
   }
 
+  const handleUpdateSnooze = async ({ notifyDate, snooze }) => {
+    setErrorMessage('')
```
Comment (ilteoood) [inline src/App.js:62]: I would put this at the end of this function, in order to avoid a double set invocation on line 65 and 72

Comment (lasmil) [reply src/App.js]: Done

**Review by ilteoood — APPROVED** [review]
LGTM

**Pull request merged**